db/blob/blob_file_reader.cc

```diff
@@ -351,6 +351,101 @@ Status BlobFileReader::GetBlob(const ReadOptions& read_options,
   return Status::OK();
 }
 
+void BlobFileReader::MultiGetBlob(
+    const ReadOptions& read_options,
+    const autovector<std::reference_wrapper<Slice>>& user_keys,
+    const autovector<uint64_t>& offsets,
+    const autovector<uint64_t>& value_sizes, autovector<Status*>& statuses,
+    autovector<PinnableSlice*>& values, uint64_t* bytes_read) const {
+  const size_t num_blobs = user_keys.size();
+  assert(num_blobs == offsets.size());
+  assert(num_blobs == value_sizes.size());
+  assert(num_blobs == statuses.size());
+  assert(num_blobs == values.size());
+
+  std::vector<FSReadRequest> read_reqs(num_blobs);
+  autovector<uint64_t> adjustments;
+  uint64_t total_len = 0;
+  for (size_t i = 0; i < num_blobs; ++i) {
+    const size_t key_size = user_keys[i].get().size();
+    assert(IsValidBlobOffset(offsets[i], key_size, value_sizes[i], file_size_));
```

Comment:
Is this currently checked by the caller (i.e. `Version`)? I think we might prefer to encapsulate/hide low-level blob file format related stuff like this in `BlobFileReader`

Reply:
@ltamasi  Yes.
Encapsulating low-level stuff in `BlobFileReader` is desirable. However, `RandomAccessFileReader::MultiRead()` requires a contiguous buffer for `FSReadRequests`. If we do not do the check in `Version`, then we need to pass `BlobIndex` info to `BlobFileReader` where it will be checked so that `BlobFileReader` will set up the buffer properly. The current implementation of `BlobFileReader` is `BlobIndex`-decoding agnostic. Consequently, I have to do the check in the `Version` layer. Let me know if you have an alternative, without having to construct intermediate vectors for offsets, sizes, and statuses.

Reply:
Yeah, that makes sense. Thanks for the explanation @riversand963 !

Reply:
I plan to add a few additional unit tests before publishing this diff.